include/crypto/evp.h

```diff
@@ -253,13 +253,46 @@ struct evp_kdf_st {
 #define EVP_ORIG_GLOBAL     1
 #define EVP_ORIG_METH       2
 
+/* Security categories for SHA algorithms */
+# define SHA1_COLLISION_CATEGORY        0
+# define SHA256_192_COLLISION_CATEGORY  0
+# define SHA224_COLLISION_CATEGORY      0
+# define SHA256_COLLISION_CATEGORY      2
+# define SHA384_COLLISION_CATEGORY      4
+# define SHA512_COLLISION_CATEGORY      5
+# define SHA3_224_COLLISION_CATEGORY    0
+# define SHA3_256_COLLISION_CATEGORY    2
+# define SHA3_384_COLLISION_CATEGORY    4
+# define SHA3_512_COLLISION_CATEGORY    5
+# define SHAKE128_COLLISION_CATEGORY    2
+# define SHAKE256_COLLISION_CATEGORY    5
+# define KMAC128_COLLISION_CATEGORY     2
+# define KMAC256_COLLISION_CATEGORY     5
+
+# define SHA1_PREIMAGE_CATEGORY         1
```

Comment:
I'm not sure...

Reply:
This value was taken from NIST's [IR 8547](https://nvlpubs.nist.gov/nistpubs/ir/2024/NIST.IR.8547.ipd.pdf) page 16.